### DISCUSSION TITLE
How to set self-hosed Plausible to boot on startup?

### DISCUSSION THREAD

**Discussion by steveharman** [plausible/analytics#818] [Self-Hosted Support] [answered]
This is more of a Docker question, but I'm not a Docker expert.  ;-)

• Ubuntu 18.0.4

How would I get my server to automatically launch Plausible [detached] when it gets rebooted please?

Thanks

**Comment by Vigasaurus** [comment, marked as answer]
Set 
```restart: unless-stopped``` (or always, if you'd like) in your docker compose, for all the services. 

See: https://docs.docker.com/compose/compose-file/compose-file-v3/#restart

**Reply by steveharman** [reply]
Thanks!  

So I could specify `restart: always` to the `docker-compose up -d` command when launching Plausible and it would know to restart all the containers on system reboot?

**Reply by Vigasaurus** [reply]
Unless stopped would do the same thing. Unless stopped basically means if you stop it with `docker stop` it won't restart automatically, but in all other scenarios it will (which is generally a good standard). But yes, either would work in this case. 

But to be clear, you need to put this in your docker-compose.yml, not as a command option. See the example that is there for the SMTP server in the given hosting repo or [stack overflow](http://stackoverflow.com/questions/42216017/ddg#43006595).
